api/ccapi/include/dataset.h

```diff
@@ -87,23 +86,28 @@ class Dataset {
 };
 
 /**
- * @brief Factory creator with constructor for dataset
+ * @copydoc std::unique_ptr<Dataset>createDataset(DatasetType, const
```

Comment:
Lets have description in the API header, and you can use copydoc in the internal headers.

Reply:
Same for other functions in this file.

Reply:
Updated :)